### ISSUE TITLE
No scrip found

### ISSUE THREAD

**Issue by sheelu987** [Kotak-Neo/Kotak-neo-api-v2#54]
import streamlit as st
import time
import pyotp
from neo_api_client import NeoAPI
import credentials as creds
from datetime import datetime

st.set_page_config(page_title="Kotak Neo Algo Pro - LIVE", layout="wide")
st.title("🤖 Kotak Neo Algo Pro - LIVE Trading (Multi Target + Re-entry)")

# Asset Configs
ASSETS_CONFIG = {
    "NIFTY": {"expiry": "10MAR2026", "lot_size": 65, "segment": "nse_fo", "product": "MIS", "strikes": list(range(23000, 26000, 50))},
    "BANKNIFTY": {"expiry": "30MAR2026", "lot_size": 30, "segment": "nse_fo", "product": "MIS", "strikes": list(range(50000, 60000, 100))},
    "SENSEX": {"expiry": "12MAR2026", "lot_size": 20, "segment": "bse_fo", "product": "MIS", "strikes": list(range(75000, 85000, 100))},
    "STOCKS": {"expiry": "", "lot_size": 1, "segment": "nse_cm", "product": "CNC", "strikes": list(range(1000, 5000, 50))}
}

# Sidebar
st.sidebar.header("🛡️ Risk & Settings")
max_profit = st.sidebar.number_input("Daily Max Profit (₹)", value=5000.0, step=100.0)
max_loss = st.sidebar.number_input("Daily Max Loss (₹)", value=-2000.0, step=-100.0)
test_mode = st.sidebar.checkbox("Test Mode (No Real Orders)", value=True)
re_entry_limit = st.sidebar.number_input("Max Re-entries", value=3, min_value=0, step=1)

st.sidebar.markdown("---")
st.sidebar.subheader("Exit Quantities")
t1_qty = st.sidebar.number_input("T1 Exit Qty", value=25, min_value=1, step=1)
t2_qty = st.sidebar.number_input("T2 Exit Qty", value=25, min_value=1, step=1)

st.sidebar.markdown("**Current Time:**")
time_placeholder = st.sidebar.empty()

# Main UI
col1, col2 = st.columns([1, 2])

with col1:
    asset_type = st.selectbox("Asset Type", list(ASSETS_CONFIG.keys()))
    config = ASSETS_CONFIG[asset_type]

    st.markdown(f"**Expiry:** {config['expiry']}")
    st.markdown(f"**Lot Size:** {config['lot_size']} | Segment: {config['segment']}")

    if asset_type == "STOCKS":
        symbol_base = st.text_input("Stock Symbol", "RELIANCE").upper()
        strike = st.selectbox("Reference Price", config["strikes"], format_func=lambda x: f"{x:,}")
        opt_type = None
        symbol = symbol_base
    else:
        symbol_base = asset_type
        strike = st.selectbox("Strike Price", config["strikes"], format_func=lambda x: f"{x:,}")
        opt_type = st.radio("Option Type", ["CE", "PE"])
        symbol = f"{symbol_base}{config['expiry']}{int(strike)}{opt_type}"

    st.info(f"**Symbol:** {symbol}")

    entry_price = st.number_input("Entry Trigger Price", value=100.0, step=0.5)
    total_qty = st.number_input("Total Quantity", value=50, min_value=1, step=1)

    if st.button("Buy at Market Price"):
        st.info("Market Buy triggered (Test Mode)")

with col2:
    st.subheader("Targets & SL")
    t1_pts = st.number_input("Target 1 Points", value=20.0, step=0.5)
    t2_pts = st.number_input("Target 2 Points", value=40.0, step=0.5)
    t3_pts = st.number_input("Target 3 Points", value=60.0, step=0.5)
    sl_pts = st.number_input("Initial SL Points", value=15.0, step=0.5)

    trail_after_t1 = st.checkbox("Trail SL to Entry after T1", value=True)
    trail_after_t2 = st.checkbox("Trail SL to T1 after T2", value=True)

# START
if st.button("🚀 START LIVE TRADING", type="primary", use_container_width=True):
    try:
        client = NeoAPI(environment=creds.ENVIRONMENT, consumer_key=creds.CONSUMER_KEY)
        totp = pyotp.TOTP(creds.TOTP_SECRET).now()
        client.totp_login(creds.MOBILE_NUMBER, creds.UCC, totp)
        client.totp_validate(creds.MPIN)
        st.success("API Connected")
    except Exception as e:
        st.error(f"Login Failed: {e}")
        st.stop()

    # Token Search (fixed format)
    inst_token = None
    try:
        search_params = {
            "exchange_segment": config["segment"],
            "symbol": symbol_base.strip()
        }

        if config["segment"] in ["nse_fo", "bse_fo", "mcx_fo"]:
            # Full year expiry for better match
            full_expiry = config["expiry"].replace("26", "2026") if "26" in config["expiry"] else config["expiry"]
            search_params.update({
                "expiry": full_expiry,
                "option_type": opt_type,
                "strike_price": str(int(strike))
            })

        st.info(f"Searching: {search_params}")

        search_result = client.search_scrip(**search_params)

        if search_result and 'data' in search_result and search_result['data']:
            inst_token = search_result['data'][0].get('token')
            st.success(f"Token: {inst_token}")
        else:
            st.warning("No scrip found - using symbol only for orders")
    except Exception as e:
        st.warning(f"Search failed: {e} - using symbol only")

    # Trading Loop (simplified for demo - add your full logic)
    current_re_entries = 0
    daily_pnl = 0.0
    is_in_trade = False
    active_sl = entry_price - sl_pts
    remaining_qty = total_qty

    status = st.empty()

    while current_re_entries < re_entry_limit:
        time_placeholder.markdown(datetime.now().strftime("%Y-%m-%d %H:%M:%S"))

        # LTP
        try:
            if inst_token:
                params = [{"instrument_token": inst_token, "exchange_segment": config["segment"]}]
                quote = client.quotes(instrument_tokens=params, quote_type="ltp")
                ltp = float(quote["data"][0]["lp"])
            else:
                ltp = entry_price + 5
        except:
            ltp = entry_price + 5

        status.metric("LIVE STATUS", f"LTP: {ltp:.2f} | SL: {active_sl if is_in_trade else '-'}", delta=f"PnL: ₹{daily_pnl:.2f}")

        time.sleep(2)

    st.error("Algo stopped.")
